x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/nulls/CoalesceTests.java

```diff
@@ -235,6 +258,55 @@ public void testCoalesceNotNullable() {
         sub.add(between(0, sub.size()), randomLiteral(sub.get(sub.size() - 1).dataType()));
         Coalesce exp = build(Source.EMPTY, sub);
         // Known not to be nullable because it contains a non-null literal
-        assertThat(exp.nullable(), equalTo(Nullability.FALSE));
+        if (testCase.expectedType() == DataType.NULL) {
+            assertThat(exp.nullable(), equalTo(Nullability.UNKNOWN));
+        } else {
+            assertThat(exp.nullable(), equalTo(Nullability.FALSE));
+        }
+    }
+
+    /**
+     * Inserts random non-null garbage <strong>around</strong> the expected data
+     * and runs
+     */
+    public void testEvaluateWithGarbage() {
```

Comment:
This is important for catching the case where your value is `null`, but the rest of the block isn't null. I had an off-by-one error in the evaluators somewhere that the standard tests weren't catching and this does.

It's not clear to me if this is worth pulling to the top level class.